### DISCUSSION TITLE
script to add a link to a page in a markdown file. how? 🤔

### DISCUSSION THREAD

**Discussion by horacioh** [johnlindquist/kit#414] [Ideas]
I'm a big [Foam](https://foambubble.github.io/foam/) user. I have a file called `inbox.md` with a huge list of links I want to read. I know use a chrome extension that copies the focused page link in markdown format (`[google](https://google.com)`) so I can just open the file and paste there at the top of the list.

I'm sure some of you'all have done something similar maybe? can someone point me to a similar example in which I can pass the link to a script and the script updates the file for me?

maybe this is a good start? https://www.scriptkit.com/johnlindquist/scripts/new-journal-entry

thanks!!

**Comment by johnlindquist** [comment]
Great question, Script Kit is _perfect_ for stuff like this:


[Install copy-tab-to-file](https://scriptkit.com/api/new?name=copy-tab-to-file&url=https://gist.githubusercontent.com/johnlindquist/0e69234077eb0d5b3effdfe7b9eb7045/raw/7fa4e7ec8428bc938ab5f099fd1acce66da60ebc/copy-tab-to-file.js")

```js
let file = home(`whatever.md`) //"home" is the same as `~/whatever.md`

let tab = await getActiveTab() //gets "url" from Chrome

// Creates or appends to file
await appendFile(
  file,
  `
[${tab}](${tab})`
)

```


We can grab the title too, although I haven't written a helper for that yet, maybe I should 🤔:


[Install copy-tab-and-title-to-file](https://scriptkit.com/api/new?name=copy-tab-and-title-to-file&url=https://gist.githubusercontent.com/johnlindquist/83457e8b4f5fa47c1bad88cade7f1901/raw/570b3a0ee57cc2a6c5cf6ded4a861d8c27fe365e/copy-tab-and-title-to-file.js")

```js
let file = home(`whatever.md`) //"home" is the same as `~/whatever.md`

//returns a comma-delimited string: foo.com, Some title
let urlAndTitle = await applescript(
  String.raw`tell application "Google Chrome" to return {URL, title} of active tab of front window`
)

let [url, title] = urlAndTitle.split(/, (.+)/) //split on first comma

// Creates or appends to file
await appendFile(
  file,
  `
[${title}](${url})`
)

```

**Comment by horacioh** [comment]
thanks @johnlindquist !

I changed it to work with Brave!

```js
let file = home(`whatever.md`) //"home" is the same as `~/whatever.md`

//returns a comma-delimited string: foo.com, Some title
let urlAndTitle = await applescript(
  String.raw`tell application "Brave" to return {URL, title} of active tab of front window`
)

let [url, title] = urlAndTitle.split(/, (.+)/) //split on first comma

// Creates or appends to file
await appendFile(
  file,
  `
- [${title}](${url})`
)

```

thanks for the help!!

**Reply by johnlindquist** [reply]
Awesome, let me know if you need any more help customizing to fit your needs. We could have it prompt for which file to save to, check if link already exists, etc, etc 😁

**Reply by horacioh** [reply]
I usually add new items at the top of the list:

```markdown
# Inbox

// <== HERE!
- [link](link)
- [link](link)
- [link](link)
- [link](link)
```

but I guess I need to use remark for this or something similar right?

**Reply by johnlindquist** [reply]
You might be overthinking it:


[Install copy-tab-and-title-to-file-middle](https://scriptkit.com/api/new?name=copy-tab-and-title-to-file-middle&url=https://gist.githubusercontent.com/johnlindquist/2a988ed594768a3d5acb05e19690da9e/raw/8995c29ee8b0d5c4bdefe4c1e1170dc63049fe55/copy-tab-and-title-to-file-middle.js")

```js
let file = home(`whatever.md`)

let urlAndTitle = await applescript(
  String.raw`tell application "Google Chrome" to return {URL, title} of active tab of front window`
)

let [url, title] = urlAndTitle.split(/, (.+)/)

let link = `[${title}](${url})`

let contents = await readFile(file, "utf-8")

let insertedLink = contents.replace(
  /(# Inbox\n)/,
  `$1
- ${link}`
)

await writeFile(file, insertedLink)
```

**Reply by horacioh** [reply]
Will try it! Thanks a lot!!

**Reply by horacioh** [reply]
worked perfectly! thanks again!!

**Comment by kenNash** [comment]
A lot of interesting stuff to me, in here! Thanks!

1. Is there a `getActiveTab()` helper for Safari, as well?
2. Did you @johnlindquist write a helper for the title (as you wrote about in your first answer) as well? For Safari?

Again, thank you!
